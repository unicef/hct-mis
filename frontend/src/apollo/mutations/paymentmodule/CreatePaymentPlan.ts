import { gql } from 'apollo-boost';

export const CREATE_PAYMENT_PLAN = gql`
  mutation CreatePP($input: CreatePaymentPlanInput!) {
    createPaymentPlan(input: $input) {
      paymentPlan {
        id
      }
    }
  }
`;
