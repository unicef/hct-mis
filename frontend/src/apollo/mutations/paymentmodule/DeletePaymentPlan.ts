import { gql } from 'apollo-boost';

export const DELETE_PAYMENT_PLAN = gql`
  mutation DeletePP($paymentPlanId: ID!) {
    deletePaymentPlan(paymentPlanId: $paymentPlanId) {
      paymentPlan {
        id
        status
      }
    }
  }
`;
