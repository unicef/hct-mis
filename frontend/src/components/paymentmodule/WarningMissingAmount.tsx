import { Box } from '@material-ui/core';
import WarningIcon from '@material-ui/icons/Warning';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

const WarnIcon = styled(WarningIcon)`
  color: ${({ theme }) => theme.hctPalette.orange};
`;

interface WarningMissingAmountProps {
  amount: number;
  currency: string;
}

export function WarningMissingAmount({
  amount,
  currency,
}: WarningMissingAmountProps): React.ReactElement {
  const { t } = useTranslation();

  return (
    <Box display='flex' alignItems='center'>
      <WarnIcon />
      {t('Missing')} {amount} {currency}
    </Box>
  );
}
