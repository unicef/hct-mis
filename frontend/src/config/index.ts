export const API_BASE_URL = `http://localhost:8082/api`;
export const GRAPHQL_URL = `${API_BASE_URL}/graphql`;
