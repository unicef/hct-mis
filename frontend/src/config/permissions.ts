export const PERMISSIONS = {
  // RDI
  RDI_VIEW_LIST: 'RDI_VIEW_LIST',
  RDI_VIEW_DETAILS: 'RDI_VIEW_DETAILS',
  RDI_IMPORT_DATA: 'RDI_IMPORT_DATA',
  RDI_RERUN_DEDUPE: 'RDI_RERUN_DEDUPE',
  RDI_MERGE_IMPORT: 'RDI_MERGE_IMPORT',

  // Population
  POPULATION_VIEW_HOUSEHOLDS_LIST: 'POPULATION_VIEW_HOUSEHOLDS_LIST',
  POPULATION_VIEW_HOUSEHOLDS_DETAILS: 'POPULATION_VIEW_HOUSEHOLDS_DETAILS',
  POPULATION_VIEW_INDIVIDUALS_LIST: 'POPULATION_VIEW_INDIVIDUALS_LIST',
  POPULATION_VIEW_INDIVIDUALS_DETAILS: 'POPULATION_VIEW_INDIVIDUALS_DETAILS',

  // User Management
  USER_MANAGEMENT_VIEW_LIST: 'USER_MANAGEMENT_VIEW_LIST',

  // Programme
  PRORGRAMME_VIEW_LIST_AND_DETAILS: 'PRORGRAMME_VIEW_LIST_AND_DETAILS',
  PROGRAMME_VIEW_PAYMENT_RECORD_DETAILS:
    'PROGRAMME_VIEW_PAYMENT_RECORD_DETAILS',
  PROGRAMME_CREATE: 'PROGRAMME_CREATE',
  PROGRAMME_UPDATE: 'PROGRAMME_UPDATE',
  PROGRAMME_REMOVE: 'PROGRAMME_REMOVE',
  PROGRAMME_ACTIVATE: 'PROGRAMME_ACTIVATE',
  PROGRAMME_FINISH: 'PROGRAMME_FINISH',

  // Targeting
  TARGETING_VIEW_LIST: 'TARGETING_VIEW_LIST',
  TARGETING_VIEW_DETAILS: 'TARGETING_VIEW_DETAILS',
  TARGETING_CREATE: 'TARGETING_CREATE',
  TARGETING_UPDATE: 'TARGETING_UPDATE',
  TARGETING_DUPLICATE: 'TARGETING_DUPLICATE',
  TARGETING_REMOVE: 'TARGETING_REMOVE',
  TARGETING_LOCK: 'TARGETING_LOCK',
  TARGETING_UNLOCK: 'TARGETING_UNLOCK',
  TARGETING_SEND: 'TARGETING_SEND',

  // Payment Verification
  PAYMENT_VERIFICATION_VIEW_LIST: 'PAYMENT_VERIFICATION_VIEW_LIST',
  PAYMENT_VERIFICATION_VIEW_DETAILS: 'PAYMENT_VERIFICATION_VIEW_DETAILS',
  PAYMENT_VERIFICATION_CREATE: 'PAYMENT_VERIFICATION_CREATE',
  PAYMENT_VERIFICATION_UPDATE: 'PAYMENT_VERIFICATION_UPDATE',
  PAYMENT_VERIFICATION_ACTIVATE: 'PAYMENT_VERIFICATION_ACTIVATE',
  PAYMENT_VERIFICATION_DISCARD: 'PAYMENT_VERIFICATION_DISCARD',
  PAYMENT_VERIFICATION_FINISH: 'PAYMENT_VERIFICATION_FINISH',
  PAYMENT_VERIFICATION_EXPORT: 'PAYMENT_VERIFICATION_EXPORT',
  PAYMENT_VERIFICATION_IMPORT: 'PAYMENT_VERIFICATION_IMPORT',
  PAYMENT_VERIFICATION_VERIFY: 'PAYMENT_VERIFICATION_VERIFY',
  PAYMENT_VERIFICATION_VIEW_PAYMENT_RECORD_DETAILS:
    'PAYMENT_VERIFICATION_VIEW_PAYMENT_RECORD_DETAILS',

  // Grievances
  GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE:
    'GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE',
  GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE_AS_CREATOR:
    'GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE_AS_CREATOR',
  GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE_AS_OWNER:
    'GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE_AS_OWNER',
  GRIEVANCES_VIEW_LIST_SENSITIVE: 'GRIEVANCES_VIEW_LIST_SENSITIVE',
  GRIEVANCES_VIEW_LIST_SENSITIVE_AS_CREATOR:
    'GRIEVANCES_VIEW_LIST_SENSITIVE_AS_CREATOR',
  GRIEVANCES_VIEW_LIST_SENSITIVE_AS_OWNER:
    'GRIEVANCES_VIEW_LIST_SENSITIVE_AS_OWNER',
  GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE:
    'GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE',
  GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE_AS_CREATOR:
    'GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE_AS_CREATOR',
  GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE_AS_OWNER:
    'GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE_AS_OWNER',
  GRIEVANCES_VIEW_DETAILS_SENSITIVE: 'GRIEVANCES_VIEW_DETAILS_SENSITIVE',
  GRIEVANCES_VIEW_DETAILS_SENSITIVE_AS_CREATOR:
    'GRIEVANCES_VIEW_DETAILS_SENSITIVE_AS_CREATOR',
  GRIEVANCES_VIEW_DETAILS_SENSITIVE_AS_OWNER:
    'GRIEVANCES_VIEW_DETAILS_SENSITIVE_AS_OWNER',
  GRIEVANCES_VIEW_HOUSEHOLD_DETAILS: 'GRIEVANCES_VIEW_HOUSEHOLD_DETAILS',
  GRIEVANCES_VIEW_HOUSEHOLD_DETAILS_AS_CREATOR:
    'GRIEVANCES_VIEW_HOUSEHOLD_DETAILS_AS_CREATOR',
  GRIEVANCES_VIEW_HOUSEHOLD_DETAILS_AS_OWNER:
    'GRIEVANCES_VIEW_HOUSEHOLD_DETAILS_AS_OWNER',
  GRIEVANCES_VIEW_INDIVIDUALS_DETAILS: 'GRIEVANCES_VIEW_INDIVIDUALS_DETAILS',
  GRIEVANCES_VIEW_INDIVIDUALS_DETAILS_AS_CREATOR:
    'GRIEVANCES_VIEW_INDIVIDUALS_DETAILS_AS_CREATOR',
  GRIEVANCES_VIEW_INDIVIDUALS_DETAILS_AS_OWNER:
    'GRIEVANCES_VIEW_INDIVIDUALS_DETAILS_AS_OWNER',
  GRIEVANCES_CREATE: 'GRIEVANCES_CREATE',
  GRIEVANCES_UPDATE: 'GRIEVANCES_UPDATE',
  GRIEVANCES_UPDATE_AS_CREATOR: 'GRIEVANCES_UPDATE_AS_CREATOR',
  GRIEVANCES_UPDATE_AS_OWNER: 'GRIEVANCES_UPDATE_AS_OWNER',
  GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE:
    'GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE',
  GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE_AS_CREATOR:
    'GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE_AS_CREATOR',
  GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE_AS_OWNER:
    'GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE_AS_OWNER',
  GRIEVANCES_ADD_NOTE: 'GRIEVANCES_ADD_NOTE',
  GRIEVANCES_ADD_NOTE_AS_CREATOR: 'GRIEVANCES_ADD_NOTE_AS_CREATOR',
  GRIEVANCES_ADD_NOTE_AS_OWNER: 'GRIEVANCES_ADD_NOTE_AS_OWNER',
  GRIEVANCES_SET_IN_PROGRESS: 'GRIEVANCES_SET_IN_PROGRESS',
  GRIEVANCES_SET_IN_PROGRESS_AS_CREATOR:
    'GRIEVANCES_SET_IN_PROGRESS_AS_CREATOR',
  GRIEVANCES_SET_IN_PROGRESS_AS_OWNER: 'GRIEVANCES_SET_IN_PROGRESS_AS_OWNER',
  GRIEVANCES_SET_ON_HOLD: 'GRIEVANCES_SET_ON_HOLD',
  GRIEVANCES_SET_ON_HOLD_AS_CREATOR: 'GRIEVANCES_SET_ON_HOLD_AS_CREATOR',
  GRIEVANCES_SET_ON_HOLD_AS_OWNER: 'GRIEVANCES_SET_ON_HOLD_AS_OWNER',
  GRIEVANCES_SEND_FOR_APPROVAL: 'GRIEVANCES_SEND_FOR_APPROVAL',
  GRIEVANCES_SEND_FOR_APPROVAL_AS_CREATOR:
    'GRIEVANCES_SEND_FOR_APPROVAL_AS_CREATOR',
  GRIEVANCES_SEND_FOR_APPROVAL_AS_OWNER:
    'GRIEVANCES_SEND_FOR_APPROVAL_AS_OWNER',
  GRIEVANCES_SEND_BACK: 'GRIEVANCES_SEND_BACK',
  GRIEVANCES_SEND_BACK_AS_CREATOR: 'GRIEVANCES_SEND_BACK_AS_CREATOR',
  GRIEVANCES_SEND_BACK_AS_OWNER: 'GRIEVANCES_SEND_BACK_AS_OWNER',
  GRIEVANCES_APPROVE_DATA_CHANGE: 'GRIEVANCES_APPROVE_DATA_CHANGE',
  GRIEVANCES_APPROVE_DATA_CHANGE_AS_CREATOR:
    'GRIEVANCES_APPROVE_DATA_CHANGE_AS_CREATOR',
  GRIEVANCES_APPROVE_DATA_CHANGE_AS_OWNER:
    'GRIEVANCES_APPROVE_DATA_CHANGE_AS_OWNER',
  GRIEVANCES_UNDO_APPROVED_DATA_CHANGE: 'GRIEVANCES_UNDO_APPROVED_DATA_CHANGE',
  GRIEVANCES_UNDO_APPROVED_DATA_CHANGE_AS_CREATOR:
    'GRIEVANCES_UNDO_APPROVED_DATA_CHANGE_AS_CREATOR',
  GRIEVANCES_UNDO_APPROVED_DATA_CHANGE_AS_OWNER:
    'GRIEVANCES_UNDO_APPROVED_DATA_CHANGE_AS_OWNER',
  GRIEVANCES_APPROVE_FLAG_AND_DEDUPE: 'GRIEVANCES_APPROVE_FLAG_AND_DEDUPE',
  GRIEVANCES_APPROVE_FLAG_AND_DEDUPE_AS_CREATOR:
    'GRIEVANCES_APPROVE_FLAG_AND_DEDUPE_AS_CREATOR',
  GRIEVANCES_APPROVE_FLAG_AND_DEDUPE_AS_OWNER:
    'GRIEVANCES_APPROVE_FLAG_AND_DEDUPE_AS_OWNER',
  GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK:
    'GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK',
  GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK_AS_CREATOR:
    'GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK_AS_CREATOR',
  GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK_AS_OWNER:
    'GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK_AS_OWNER',
  GRIEVANCES_CLOSE_TICKET_FEEDBACK: 'GRIEVANCES_CLOSE_TICKET_FEEDBACK',
  GRIEVANCES_CLOSE_TICKET_FEEDBACK_AS_CREATOR:
    'GRIEVANCES_CLOSE_TICKET_FEEDBACK_AS_CREATOR',
  GRIEVANCES_CLOSE_TICKET_FEEDBACK_AS_OWNER:
    'GRIEVANCES_CLOSE_TICKET_FEEDBACK_AS_OWNER',
};

export function hasPermissions(
  permission: string | string[],
  allowedPermissions: string[],
): boolean {
  // checks to see if has one permission or at least one from the array

  if (Array.isArray(permission)) {
    return allowedPermissions.some((perm) => permission.includes(perm));
  }
  return allowedPermissions.includes(permission);
}

export function hasPermissionInModule(
  module: string,
  allowedPermissions: string[],
): boolean {
  return allowedPermissions.some((perm) => perm.includes(module));
}
