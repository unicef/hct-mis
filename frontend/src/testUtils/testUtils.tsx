import React, { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import cloneDeep from 'lodash/cloneDeep';
import { TestProviders } from './testProviders';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(<BrowserRouter>{ui}</BrowserRouter>, {
    wrapper: TestProviders,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export const makeApolloLoadingMock = (mocks) => {
  let mocksWithDelay = cloneDeep(mocks);
  for (const mock of mocksWithDelay) {
    mock.request.delay = 1e21;
  }
  return mocksWithDelay;
};
