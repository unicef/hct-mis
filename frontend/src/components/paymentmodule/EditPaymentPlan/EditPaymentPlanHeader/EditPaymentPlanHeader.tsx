import { Box, Button } from '@material-ui/core';
import { Link } from 'react-router-dom';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { hasPermissions, PERMISSIONS } from '../../../../config/permissions';
import { BreadCrumbsItem } from '../../../core/BreadCrumbs';
import { PageHeader } from '../../../core/PageHeader';
import { PaymentPlanQuery } from '../../../../__generated__/graphql';
import { StatusBox } from '../../../core/StatusBox';
import {
  paymentPlanBackgroundActionStatusMapping,
  paymentPlanBackgroundActionStatusToColor,
  paymentPlanStatusMapping,
  paymentPlanStatusToColor,
} from '../../../../utils/utils';

const StatusWrapper = styled.div`
  width: 140px;
  margin-left: 30px;
`;

interface EditPaymentPlanHeaderProps {
  handleSubmit: () => Promise<void>;
  businessArea: string;
  permissions: string[];
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const EditPaymentPlanHeader = ({
  handleSubmit,
  businessArea,
  permissions,
  paymentPlan,
}: EditPaymentPlanHeaderProps): React.ReactElement => {
  const { t } = useTranslation();
  const { id } = paymentPlan;

  const breadCrumbsItems: BreadCrumbsItem[] = [
    {
      title: t('Payment Module'),
      to: `/${businessArea}/payment-module/payment-plans/${id}`,
    },
  ];

  return (
    <PageHeader
      title={
        <Box display='flex' alignItems='center'>
          {t('Payment Plan')} ID {paymentPlan.unicefId}
          <StatusWrapper>
            <StatusBox
              status={paymentPlan.status}
              statusToColor={paymentPlanStatusToColor}
              statusNameMapping={paymentPlanStatusMapping}
            />
          </StatusWrapper>
          {paymentPlan.backgroundActionStatus && <StatusWrapper>
              <StatusBox
                  status={paymentPlan.backgroundActionStatus}
                  statusToColor={paymentPlanBackgroundActionStatusToColor}
                  statusNameMapping={paymentPlanBackgroundActionStatusMapping}
              />
          </StatusWrapper>}
        </Box>
      }
      breadCrumbs={
        hasPermissions(PERMISSIONS.PAYMENT_MODULE_VIEW_LIST, permissions)
          ? breadCrumbsItems
          : null
      }
    >
      <Box display='flex' mt={2} mb={2}>
        <Box mr={3}>
          <Button
            component={Link}
            to={`/${businessArea}/payment-module/payment-plans/${id}`}
          >
            {t('Cancel')}
          </Button>
        </Box>
        <Button variant='contained' color='primary' onClick={handleSubmit}>
          {t('Save')}
        </Button>
      </Box>
    </PageHeader>
  );
};
