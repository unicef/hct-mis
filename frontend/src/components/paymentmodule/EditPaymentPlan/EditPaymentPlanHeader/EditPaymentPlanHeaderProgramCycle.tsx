import { Box, Button } from '@material-ui/core';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { hasPermissions, PERMISSIONS } from '../../../../config/permissions';
import {
  getPaymentPlanUrlPart,
  paymentPlanBackgroundActionStatusToColor,
  paymentPlanStatusToColor,
} from '../../../../utils/utils';
import { PaymentPlanQuery } from '../../../../__generated__/graphql';
import { BreadCrumbsItem } from '../../../core/BreadCrumbs';
import { PageHeader } from '../../../core/PageHeader';
import { StatusBox } from '../../../core/StatusBox';

const StatusWrapper = styled.div`
  width: 140px;
  margin-left: 30px;
`;

interface EditPaymentPlanHeaderProgramCycleProps {
  handleSubmit: () => Promise<void>;
  baseUrl: string;
  permissions: string[];
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

export const EditPaymentPlanHeaderProgramCycle = ({
  handleSubmit,
  baseUrl,
  permissions,
  paymentPlan,
}: EditPaymentPlanHeaderProgramCycleProps): React.ReactElement => {
  const { t } = useTranslation();
  const { id, isFollowUp } = paymentPlan;

  const breadCrumbsItems: BreadCrumbsItem[] = [
    {
      title: t('Payment Module'),
      to: `/${baseUrl}/payment-module/${getPaymentPlanUrlPart(
        isFollowUp,
      )}/${id}`,
    },
    {
      title: `${paymentPlan.programCycle.name} (ID: ${paymentPlan.programCycle.unicefId})`,
      to: `/${baseUrl}/payment-module/program-cycles/${paymentPlan.programCycle.id}`,
    },
  ];

  return (
    <PageHeader
      title={
        <Box display='flex' alignItems='center'>
          {t(isFollowUp ? 'Follow-up Payment Plan' : 'Payment Plan')} ID{' '}
          {paymentPlan.unicefId}
          <StatusWrapper>
            <StatusBox
              status={paymentPlan.status}
              statusToColor={paymentPlanStatusToColor}
            />
          </StatusWrapper>
          {paymentPlan.backgroundActionStatus && (
            <StatusWrapper>
              <StatusBox
                status={paymentPlan.backgroundActionStatus}
                statusToColor={paymentPlanBackgroundActionStatusToColor}
              />
            </StatusWrapper>
          )}
        </Box>
      }
      breadCrumbs={
        hasPermissions(PERMISSIONS.PM_VIEW_LIST, permissions)
          ? breadCrumbsItems
          : null
      }
    >
      <Box display='flex' mt={2} mb={2}>
        <Box mr={3}>
          <Button
            component={Link}
            to={`/${baseUrl}/payment-module/program-cycles/${
              paymentPlan.programCycle.id
            }/${getPaymentPlanUrlPart(isFollowUp)}/${id}`}
          >
            {t('Cancel')}
          </Button>
        </Box>
        <Button variant='contained' color='primary' onClick={handleSubmit}>
          {t('Save')}
        </Button>
      </Box>
    </PageHeader>
  );
};
