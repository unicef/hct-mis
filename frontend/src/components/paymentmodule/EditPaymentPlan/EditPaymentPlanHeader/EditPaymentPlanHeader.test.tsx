import React from 'react';
import { fakeApolloPaymentPlan } from '../../../../../fixtures/paymentmodule/fakeApolloPaymentPlan';
import { PERMISSIONS } from '../../../../config/permissions';
import { render } from '../../../../testUtils/testUtils';
import { EditPaymentPlanHeader } from './EditPaymentPlanHeader';

describe('components/paymentmodule/EditPaymentPlanHeader', () => {
  it('should render', () => {
    const { container } = render(
      <EditPaymentPlanHeader
        handleSubmit={jest.fn()}
        businessArea='afghanistan'
        permissions={[PERMISSIONS.PM_PAYMENT_MODULE_VIEW_LIST]}
        paymentPlan={fakeApolloPaymentPlan}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
