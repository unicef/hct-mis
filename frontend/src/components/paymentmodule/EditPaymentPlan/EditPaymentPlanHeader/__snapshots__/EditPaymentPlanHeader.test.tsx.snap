// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/EditPaymentPlanHeader should render 1`] = `
<div>
  <div
    class="sc-bxivhb ixnywI"
    data-cy="page-header-container"
  >
    <div
      class="sc-ifAKCX dWpFEd"
    >
      <div
        class="sc-gzVnrw btWNie"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 11H7.83l4.88-4.88c.39-.39.39-1.03 0-1.42a.9959.9959 0 00-1.41 0l-6.59 6.59c-.39.39-.39 1.02 0 1.41l6.59 6.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L7.83 13H19c.55 0 1-.45 1-1s-.45-1-1-1z"
          />
        </svg>
      </div>
      <div
        class="sc-EHOje PMFav"
      >
        <div>
          <div
            class="sc-htpNat kquAIx"
          >
            <span
              class="sc-bwzfXH fNOhzS"
            >
              <a
                class="sc-bdVaJa eFfWno"
                href="/afghanistan/payment-module/payment-plans/UGF5bWVudFBsYW5Ob2RlOmE5YzJjMmM4LWJmYWUtNDBhMy05YmYwLWIxYWE1ZmRlMDE0YQ=="
              >
                Payment Module
              </a>
            </span>
          </div>
          <div
            class="MuiBox-root MuiBox-root-1"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
            >
              <div
                class="MuiBox-root MuiBox-root-2"
              >
                Payment Plan
                 ID 
                PP-0060-22-00000001
                <div
                  class="sc-gqjmRU bKymfA"
                >
                  <div
                    class="sc-iwsKbI dUmvrG"
                  >
                    <div
                      class="sc-gZMcBi iKWapA"
                      data-cy="status-container"
                    >
                      In Authorization
                    </div>
                  </div>
                </div>
              </div>
            </h5>
            <div
              class="MuiBox-root MuiBox-root-3"
            />
          </div>
        </div>
        <div
          class="sc-bZQynM jdkaLI"
        >
          <div
            class="MuiBox-root MuiBox-root-4"
          >
            <div
              class="MuiBox-root MuiBox-root-5"
            >
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                href="/afghanistan/payment-module/payment-plans/UGF5bWVudFBsYW5Ob2RlOmE5YzJjMmM4LWJmYWUtNDBhMy05YmYwLWIxYWE1ZmRlMDE0YQ=="
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  Cancel
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Save
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-htoDjs hzwHVL"
    />
  </div>
</div>
`;
